### DISCUSSION TITLE
DAB radio     rAudio 20230212

### DISCUSSION THREAD

**Discussion by yves2** [rern/rAudio#1237] [Bug Reports]
when i click on the test 
i only get:
rtsp-simple-server 0.21.2-1
# rtl_test -t
Found 1 device(s):
  0:  Realtek, RTL2838UHIDIR, SN: 00000001

Using device 0: Generic RTL2832U OEM
Found Rafael Micro R820T/2 tuner
Supported gain values (29): 0.0 0.9 1.4 2.7 3.7 7.7 8.7 12.5 14.4 15.7 16.6 19.7 20.7 22.9 25.4 28.0 29.7 32.8 33.8 36.4 37.2 38.6 40.2 42.1 43.4 43.9 44.5 48.0 49.6 
Sampling at 2048000 S/s.
Testing tuner range. This may take a couple of minutes..

and nothing more
whis ssh connection the command 
rAudio:~  #  dab-scanner-rtlsdr -C 5A
find all the radios available

rAudio:~  #  dab-scanner-rtlsdr -C 5A
dab_scanner V 1.0alfa,
Copyright 2018 J van Katwijk, Lazy Chair Computing
2018 Hayati Ayguen
2019 J van Katwijk
2020 J van Katwijk
going for rtlsdr 174928000 50
OK, functions seem to be loaded
Found Rafael Micro R820T/2 tuner
samplerate set to 2048000
Supported gain values (29): 0.0 0.9 1.4 2.7 3.7 7.7 8.7 12.5 14.4 15.7 16.6 19.7 20.7 22.9 25.4 28.0 29.7 32.8 33.8 36.4 37.2 38.6 40.2 42.1 43.4 43.9 44.5 48.0 49.6 
effective gain: gain 25.4
Allocating 15 zero-copy buffers
checking data in channel 5A
Allocating 15 zero-copy buffers
dab processor will stop
checking data in channel 5B
Allocating 15 zero-copy buffers
dab processor will stop
checking data in channel 5C
Allocating 15 zero-copy buffers
dab processor will stop
checking data in channel 5D
Allocating 15 zero-copy buffers
dab processor will stop
checking data in channel 6A
ensemble PARIS 6A is (F00E) recognized

Ensemble PARIS 6A; ensembleId F00E; channel 6A; 

Audio services
type;program name;channel;serviceId;subchannelId;start address;length (CU); bit rate;DAB/DAB+; genre; prot level; code rate; language

.......................................

**Comment by yves2** [comment]
finally get the dab to work
on raspberry os with welle.io
the problem is that my dabstick need a correction of 68ppm
back to raudio i tried :
dab-rtlsdr-3 -C 8C -S F202 -p68 -i /tmp/|aplay -f S16_LE -r 48000 -t raw -c 2
and it works so now i wonder what to do

**Comment by vale-max** [comment]
@yves2 you can try editing file /srv/http/bash/dab-start.sh changing line 15 from
`dab-rtlsdr-3 \`
to
`dab-rtlsdr-3 -p68 \`

**Reply by yves2** [reply]
it works  
after also adding -p68  in the second line of
 /srv/http/bash/dab-scan.sh
script -c 'dab-scanner-rtlsdr -p68 -C 5A'
the scan works
thanks

**Reply by Lightwel** [reply]
Hi
 
# rtl_test -t 

Found 1 device(s):
  0:  Realtek, RTL2838UHIDIR, SN: 00000001

Using device 0: Generic RTL2832U OEM
Found Rafael Micro R820T/2 tuner
Supported gain values (29): 0.0 0.9 1.4 2.7 3.7 7.7 8.7 12.5 14.4 15.7 16.6 19.7 20.7 22.9 25.4 28.0 29.7 32.8 33.8 36.4 37.2 38.6 40.2 42.1 43.4 43.9 44.5 48.0 49.6 
Sampling at 2048000 S/s.
Testing tuner range. This may take a couple of minutes..

and nothing.

# dab-scanner-rtlsdr -C 5A
dab_scanner V 1.0alfa,
Copyright 2018 J van Katwijk, Lazy Chair Computing
2018 Hayati Ayguen
2019 J van Katwijk
2020 J van Katwijk
going for rtlsdr 174928000 50
OK, functions seem to be loaded
Found Rafael Micro R820T/2 tuner
samplerate set to 2048000
Supported gain values (29): 0.0 0.9 1.4 2.7 3.7 7.7 8.7 12.5 14.4 15.7 16.6 19.7 20.7 22.9 25.4 28.0 29.7 32.8 33.8 36.4 37.2 38.6 40.2 42.1 43.4 43.9 44.5 48.0 49.6 
effective gain: gain 25.4

Sample output from dab-scanner 

audioservice;Jazz FM Stereo  ;11A;CAD7;17;436;24;32;DAB+;24;EEP 3-A;1/2;unknown;

or 

audioservice;BBC Radio 4     ;12B;C224;4;308;96;128;DAB;9;UEP 3;3/5;English;

How did you find the required correction "-p68" ? and did you have to do the amendments to both /srv/http/bash/dab-start.sh and /srv/http/bash/dab-scan.sh?

Tks
R

**Reply by yves2** [reply]
my dabstick was working well on my desktop ,
but not on the raspberry with raudio so i tried
 with raspberry pi os  and welle.io (in the repository,there is no package for Qt-dab)
there was a correction around 68 for all the transmitters
so  have done correction on both /srv/http/bash/dab-start.sh and /srv/http/bash/dab-scan.sh

**Comment by Lightwel** [comment]
I see. The error correction required on my carousels is about -1.7 ppm but varies between 0.7 and 1.9.Do you think rAudio could implement similar auto-correction to welle.io?BestRSent from my iPhoneOn 26 Feb 2023, at 01:00, yves2 ***@***.***> wrote:﻿
my dabstick was working well on my desktop ,
but not on the raspberry with raudio so i tried
with raspberry pi os  and welle.io (in the repository,there is no package for Qt-dab)
there was a correction around 68 for all the transmitters
so  have done correction on both /srv/http/bash/dab-start.sh and /srv/http/bash/dab-scan.sh

**Comment by Lightwel** [comment]
Not working for me...

Amended dab-start.sh and dab-scan.sh by adding -p2, then when that did nothing -p1

rAudio:/srv/http/bash  #  /srv/http/bash/dab-start.sh

dab_cmdline example III,
 	                  Copyright 2017 J van Katwijk, Lazy Chair Computing
options are i:T:D:d:M:B:P:O:A:C:G:p:Q:S:
going for rtlsdr 220352000 50
OK, functions seem to be loaded
Found Rafael Micro R820T/2 tuner
samplerate set to 2048000
Supported gain values (29): 0.0 0.9 1.4 2.7 3.7 7.7 8.7 12.5 14.4 15.7 16.6 19.7 20.7 22.9 25.4 28.0 29.7 32.8 33.8 36.4 37.2 38.6 40.2 42.1 43.4 43.9 44.5 48.0 49.6 
effective gain: gain 25.4
Allocating 15 zero-copy buffers

rAudio:/srv/http/bash  #  /srv/http/bash/dab-scan.sh

hangs ..

^C 
rm: cannot remove '/srv/http/data/shm/updatingdab': No such file or directory

This looks as if the scan returns nothing at all.  

However,

rAudio:~  #  dab-scanner-rtlsdr -C 10A
dab_scanner V 1.0alfa,
Copyright 2018 J van Katwijk, Lazy Chair Computing
2018 Hayati Ayguen
2019 J van Katwijk
2020 J van Katwijk
going for rtlsdr 209936000 50
OK, functions seem to be loaded
Found Rafael Micro R820T/2 tuner
samplerate set to 2048000
Supported gain values (29): 0.0 0.9 1.4 2.7 3.7 7.7 8.7 12.5 14.4 15.7 16.6 19.7 20.7 22.9 25.4 28.0 29.7 32.8 33.8 36.4 37.2 38.6 40.2 42.1 43.4 43.9 44.5 48.0 49.6 
effective gain: gain 25.4
Allocating 15 zero-copy buffers
checking data in channel 10A

produces a list of stations.

Completely lost here .. an example of the raw information from welle-io is

BBC Radio 1

Current channel: 12B (225.648 MHz)
Frequency correction: 225Hz (-1.14 ppm)
SNR: 8.8 dB
Frame Sync: OK
FIC CRC: Error
Frame errors: 0

 
Any help greatly appreciated.

BTW every time I have to reboot the pi I have to unplug and reconnect the dab-stick.  Is there a fix?

Tks
R
